### PR TITLE
[improve][doc] added instructions about how to run ci from fork

### PR DESCRIPTION
**Pull request by heesung-sn**
### Motivation
As commented [here](https://github.com/apache/pulsar/pull/15264#issuecomment-1113076698), moving this instruction, `how to run Pulsar CI from fork`, to the contribution page in order to make it more accessible for new contributors.

### Modifications

Added instructions about how to run Pulsar CI from forked github repositories in the contribution page. 

### Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API: (yes / **no**)
  - The schema: (yes / **no** / don't know)
  - The default values of configurations: (yes / **no**)
  - The wire protocol: (yes / **no**)
  - The rest endpoints: (yes / **no**)
  - The admin cli options: (yes / **no**)
  - Anything that affects deployment: (yes / **no** / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `no-need-doc` 
(Please explain why)
  
- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-added`
(Docs have been already added)

### CODE REVIEW

File: site2/website/contributing.md [hunk 1]
```diff
@@ -149,6 +149,97 @@ Then you can push your local, committed changes to your (forked) repository on G
...
+```
+- name: Setup ssh access to build runner VM
+  # ssh access is enabled for builds in own forks
+  if: ${{ github.repository != 'apache/pulsar' && needs.changed_files_job.outputs.docs_only != 'true' }}
+  uses: ./.github/actions/ssh-access
+  with:
+  limit-access-to-actor: true
+```
+and inline composite action [implementation](https://github.com/apache/pulsar/blob/master/.github/actions/ssh-access/action.yml).
+The ssh acccess is secured with the ssh key registered in Github.
```
Comment (momo-jun) [inline site2/website/contributing.md:239]:
```suggestion
The ssh access is secured with the ssh key registered in GitHub.
```

File: site2/website/contributing.md [hunk 2]
```diff
@@ -149,6 +149,97 @@ Then you can push your local, committed changes to your (forked) repository on G
...
+For example, your public keys are https://github.com/horizonzy.keys.
+You will first have to register a ssh public key in GitHub for that to work.
```
Comment (momo-jun) [inline site2/website/contributing.md:241]:
```suggestion
You will first have to register an ssh public key in GitHub for that to work.
```

File: site2/website/contributing.md [hunk 3]
```diff
@@ -149,6 +149,97 @@ Then you can push your local, committed changes to your (forked) repository on G
...
+
+Install GitHub CLI from https://cli.github.com/ and configure it.
+With GH CLI, there's an easy way to open the PR to your own fork with a
+single command:
+```
+gh pr create --repo=your_github_id/pulsar --base master --head your-pr-branch -f
+```
+B.) You can also create a PR to your own fork in the GitHub UI when opening
+a new PR. To do so, first click on "compare across forks" and then choose
+your own fork as both the fork repository and head repository.
```
Comment (momo-jun) [inline site2/website/contributing.md:193]:
```suggestion
your own fork as both the forked repository and head repository.
```

File: site2/website/contributing.md [hunk 4]
```diff
@@ -149,6 +149,97 @@ Then you can push your local, committed changes to your (forked) repository on G
...
+
+#### Stay in-sync with apache/pulsar:master
+
+It's worth keeping your master branch in sync with apache/pulsar's master
+so that the PR diff will be reasonable in your own fork.
+
+Here's one way to sync your fork's master branch with apache/pulsar's
+master branch:
+Let's say if you have git remotes called "upstream" for apache/pulsar and
```
Comment (momo-jun) [inline site2/website/contributing.md:202]:
```suggestion
Let's say you have git remotes called "upstream" for apache/pulsar and
```

File: site2/website/contributing.md (hunk 1 above)
Comment (Anonymitaet) [reply site2/website/contributing.md:239]: +1

and "SSH, SSH key", please pay attention to the capitalization for proper nouns

**Comment by heesung-sn** [comment]
Thank you for checking this PR. I resolved the above comments.

File: site2/website/contributing.md (hunk 3 above)
Comment (heesung-sn) [reply site2/website/contributing.md]: updated.

File: site2/website/contributing.md (hunk 2 above)
Comment (heesung-sn) [reply site2/website/contributing.md]: updated.

File: site2/website/contributing.md (hunk 1 above)
Comment (heesung-sn) [reply site2/website/contributing.md]: updated.

**Review by momo-jun — APPROVED** [review]

**Review by Anonymitaet — APPROVED** [review]

**Comment by lhotari** [comment]
@heesung-sn @Anonymitaet It looks like these instructions have been dropped and aren't on the new website. Would it be possible to get the content to the new website too?

**Comment by heesung-sn** [comment]
> @heesung-sn @Anonymitaet It looks like these instructions have been dropped and aren't on the new website. Would it be possible to get the content to the new website too?

Raised a PR to the new website repo, https://github.com/apache/pulsar-site/pull/111/ 
@Anonymitaet could you check this by any chance?

**Comment by Anonymitaet** [comment]
> > @heesung-sn @Anonymitaet It looks like these instructions have been dropped and aren't on the new website. Would it be possible to get the content to the new website too?
> 
> Raised a PR to the new website repo, [apache/pulsar-site#111](https://github.com/apache/pulsar-site/pull/111) @Anonymitaet could you check this by any chance?

sure, commented in your new PR.

**Pull request merged**